### PR TITLE
Added support for LinearClassifier, LinearRegressor and Normalizer

### PR DESCRIPTION
**Pull request by threeseed**
* Adds support for ONNX LinearClassifier, LinearRegressor, Normalizer ML operations.
* Adds benchmarks for LinearClassifier, LinearRegressor.

Verified across dozens of models.

### CODE REVIEW

File: onnx/src/ops/ml/linear_classifier.rs [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+                tract_core::ops::cast::cast(f32::datum_type()),
+                &[x],
+            )?[0];
+        }
+
+        let w = model.add_const(format!("{prefix}.coefficients"), self.coefficients.clone())?;
+        let mut scores = {
+            model.wire_node(
+                format!("{prefix}.matmul"),
+                PrefixMatMul {
```
Comment (kali) [inline onnx/src/ops/ml/linear_classifier.rs]: Mmm... this will not play well with the optimiser. PrefixMatMul is meant to be an intermediate step in converting tract-core back to nnef or tflite, or to some optimisations.
tract-core favours and einsum reprensentation for matmul, as it gives the framework maximum flexibility for axis reordering.
The consequence is, this PrefixMatMul will not be optimised at all on cpu.
Fixing should not be too hard, look at this couple of lines for inspiration:
https://github.com/sonos/tract/blob/main/onnx/src/ops/math/gemm.rs#L61C1-L66C15

File: onnx/src/ops/ml/linear_classifier.rs [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+                        },
+                        &logits2,
+                    )?
+                }
+            }
+        } else {
+            let mut tmp = scores.clone();
+            match self.post_transform {
+                None => {}
+                Some(PostTransform::Softmax) => {
```
Comment (kali) [inline onnx/src/ops/ml/linear_classifier.rs]: Mmm... instantiating Softmax is not easy :)
how about using `Softmax  { axes: tvec!(1), ..Softmax::default() }` to avoid exposing this code to this complexity ?

File: onnx/src/ops/ml/linear_regressor.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                format!("{prefix}.to_f32"),
+                tract_core::ops::cast::cast(f32::datum_type()),
+                &[x],
+            )?[0];
+        }
+
+        let w = model.add_const(format!("{prefix}.coefficients"), self.coefficients.clone())?;
+        let mut y = model.wire_node(
+            format!("{prefix}.matmul"),
+            PrefixMatMul {
```
Comment (kali) [inline onnx/src/ops/ml/linear_regressor.rs]: Einsum again

File: onnx/Cargo.toml [hunk 4]
```diff
@@ -24,18 +24,29 @@ log.workspace = true
 memmap2.workspace = true
 num-integer.workspace = true
 prost.workspace = true
+rand.workspace = true
+rayon.workspace = true
```
Comment (kali) [inline onnx/Cargo.toml]: same here, right ?

**Review by kali — CHANGES_REQUESTED** [review]
